api/net/SignalR/MessageHub.cs

```diff
@@ -14,19 +14,55 @@ public class MessageHub : Hub
 {
     #region Variables
     private readonly JsonSerializerOptions _serializerOptions;
+    private readonly ILogger _logger;
     #endregion
 
     #region Constructors
     /// <summary>
     /// Creates a new instance of a MessageHub, initializes with specified parameters.
     /// </summary>
     /// <param name="serializerOptions"></param>
-    public MessageHub(IOptions<JsonSerializerOptions> serializerOptions)
+    /// <param name="logger"></param>
+    public MessageHub(IOptions<JsonSerializerOptions> serializerOptions, ILogger<MessageHub> logger)
     {
         _serializerOptions = serializerOptions.Value;
+        _logger = logger;
     }
     #endregion
 
     #region Methods
+    /// <summary>
+    /// Allows a user to join a group for messages.
+    /// </summary>
+    /// <param name="name"></param>
+    /// <returns></returns>
+    [HubMethodName("JoinRoom")]
+    public async Task JoinRoomAsync(string name)
```

Comment:
We now have a simple concept of message rooms.  We can add security to these when we start sending information that shouldn't be shared.